cluster-agent/utils/argocd_login_credentials_test.go

```diff
@@ -2,118 +2,108 @@ package utils
-
-	// Acquire from the cache
-	creds, argoClient, err = cs.GetArgoCDLoginCredentials(context.Background(), "openshift-gitops", "12", false, k8sClient)
-	assert.NoError(t, err)
-	assert.NotEmpty(t, creds)
-	assert.NotEmpty(t, argoClient)
-
-}
+var _ = Describe("ArgoCD Login Credentials", func() {
+	Context("ArgoCD Login Credentials Test", func() {
+		It("Get ArgoCD LoginCredentials to retrieve the login credentials for an Argo CD cluster", func() {
+			var err error
+
+			var k8sClient client.Client
+			{
+
+				namespace := &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "openshift-gitops",
+						Namespace: "openshift-gitops",
+					},
+				}
+
+				loginSecret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "openshift-gitops-cluster",
+						Namespace: "openshift-gitops",
+					},
+					Data: map[string][]byte{
+						"admin.password": []byte("a1Y4c0RvdkgxcHFPUTNJYWxSaDRubXlaZ3c3QUJGcmQ="),
+					},
+				}
+
+				route := &routev1.Route{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "openshift-gitops-server",
+						Namespace: "openshift-gitops",
+					},
+					Spec: routev1.RouteSpec{
+						Port: &routev1.RoutePort{
+							TargetPort: intstr.FromString("https"),
+						},
+						Host: "route-host",
+					},
+					Status: routev1.RouteStatus{},
+				}
+
+				k8sClient, err = generateFakeK8sClient(namespace, loginSecret, route)
+				Expect(err).To(BeNil())
+			}
+
+			mockVersionClient := &mocks.VersionServiceClient{}
+			mockSessionClient := &mocks.SessionServiceClient{}
+			mockAppClient := &mocks.Client{}
+
+			mockAppClient.On("NewSettingsClient").Return(mockCloser{}, nil, nil)
+			mockAppClient.On("NewSessionClient").Return(mockCloser{}, mockSessionClient, nil)
+
+			By("The code being tested only expects a version client that returns a non-error")
+			mockAppClient.On("NewVersionClient").Return(mockCloser{}, mockVersionClient, nil)
+			mockVersionClient.On("Version", mock.Anything, mock.Anything).Return(&version.VersionMessage{Version: ""}, nil)
+
+			mockSessionClient.On("Create", mock.Anything, &session.SessionCreateRequest{
+				Username: "admin",
+				Password: "a1Y4c0RvdkgxcHFPUTNJYWxSaDRubXlaZ3c3QUJGcmQ=",
+				Token:    "",
+			}).Return(&session.SessionResponse{Token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDUyMDM1NDksImp0aSI6ImM0Nzc2MDgwLTk3NDgtNGRhNS1iODU5LWMwM2QyM2Y3MGZhMiIsImlhdCI6MTY0NTExNzE0OSwiaXNzIjoiYXJnb2NkIiwibmJmIjoxNjQ1MTE3MTQ5LCJzdWIiOiJhZG1pbjpsb2dpbiJ9.7Di4Eb7xdBEF6SiScWbM5JdwE2Z_Kgr2hfzYA-KSLFs"}, nil)
+
+			clientGenerator := mockClientGenerator{
+				mockClient: mockAppClient,
+			}
+
+			cs := NewCredentialService(&clientGenerator, true)
+			Expect(err).To(BeNil())
+
+			creds, argoClient, err := cs.GetArgoCDLoginCredentials(context.Background(), "openshift-gitops", "12", true, k8sClient)
+			Expect(err).To(BeNil())
+			Expect(argoCDCredentials{} == creds).To(BeFalse())
+			Expect(argoClient).ToNot(BeNil())
```

Comment:
You are supposed to check for empty object here as well.

Reply:
For argoClient ? @jparsai

Reply:
Yes, in existing test I see empty check for argoClient as well.
https://github.com/redhat-appstudio/managed-gitops/blob/2fe8209b9711418db191393fc601eea2fbd6f2df/cluster-agent/utils/argocd_login_credentials_test.go#L102

Reply:
argoClient is "var argoClient argocdclient.Client"  here argocdclient.Client is an interface, so we cannot check for empty object @jparsai

Reply:
Okay, I see in case of failure `GetArgoCDLoginCredentials` returns `nil` for `argoClient`, so I guess no need to check for empty like existing test just check for `nil` is enough.
https://github.com/redhat-appstudio/managed-gitops/blob/2fe8209b9711418db191393fc601eea2fbd6f2df/cluster-agent/utils/argocd_login_credentials.go#L59

Reply:
Yup @jparsai  then there is no need to change this "Expect(argoClient).ToNot(BeNil())" right ?